cache_test.go

```diff
@@ -133,7 +133,7 @@ func TestNewCache(t *testing.T) {
 		MaxCost:     10,
 		BufferItems: 0,
 	})
-	require.Error(t, err)
```

Comment:
Is it possible to add more test cases pertinent to the changes in defaulting behavior? This is to make sure that any future changes to this behavior are rightly caught by the test case(s)

Reply:
I just tried making these tests a bit more explicit. PTAL

I don't think we can significantly improve coverage further here: we can't realistically check for all cases where  we _don't_ expect an error, so we'd bound to covering the error cases. WDYT?